csharp/src/SeedLang/Runtime/NumberRange.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace SeedLang.Runtime {
+  // A class to store the range information of "for in range" statements.
+  internal class NumberRange {
```

Comment:
Naming: python simply names this type `range`. Any particular reason you named it `NumberRange`?

Reply:
There are conflicts with Range class in Common component if naming it as Range, we have to write a lot of code like Common.Range. I'm thinking about different options:
 
1. Change the name to RangeValue or something else better.
2. Add a namespace SeedLang.Value and put all value types include Range in it. There are so many value types now. But we still need write Value.Range in the code.
3. Add a namespace SeedLang.Runtime.Value and put all value types include Range in it. There are so many value types now. But we still need write Value.Range in the code.
 
Any ideas about it? Which one do you prefer?

Reply:
Ah, good question. I forgot the class name of the source code range.

### Why is `NumberRange` not good?

The name `NumberRange` is leading to an assumption that there are other range classes, such as `StringRange`, `BoolRange`, etc., which do not and will not exist.

### My preference

The 1st idea looks more promising to me. How about a set of Value classes like:

* NumberValue
* StringValue
* BoolValue
* ListValue
* IterValue
* RangeValue
* ...

I am not sure if this naming system is conflicting with your current design.

And in parallel, we might need to have another set of Type classes, like:

* NumberType
* StiringType
* BoolType
* ListType
* IterType
* RangeType
* ...

The could be consistent and easy to understand.

Reply:
A little problem here, current design is we only have one "struct Value" to hole None, bool, number and HeapObject. The HeapObject will then hole String, Function, List, Iter, Range.

String and List are just C# host objects, Function, Iter and Range are customized heap objects. I think it's not good to define them as XXXObject.

I have a idea that is to define these customized heap objects in the HeapObject class. Let me try if it's a good idea.

Reply:
Moved Function, Range into HeapObject class.